keras/02-the-layer-class.md

```diff
@@ -0,0 +1,317 @@
+## The `Layer` class
+
+([Source](https://github.com/keras-team/keras/blob/r2.6/keras/engine/base_layer.py#L84))
+
+The `Layer` class is easy to understand. It is the base class for the neural
+network layers in Keras.
+
+Here is a simple example of inheriting the `Layer` class to build a custom
+layer.
+
+```py
+from tensorflow import keras
+
+class SimpleDense(keras.layers.Layer):
+  def __init__(self, units=32):
+      super(SimpleDense, self).__init__()
+      self.units = units
+  def build(self, input_shape):
+      self.w = self.add_weight(shape=(input_shape[-1], self.units),
+                               initializer='random_normal',
+                               trainable=True)
+      self.b = self.add_weight(shape=(self.units,),
+                               initializer='random_normal',
+                               trainable=True)
+  def call(self, inputs):
+      return tf.matmul(inputs, self.w) + self.b
+```
+
+From this example, we can see that a `Layer` instance is a collection of
```

Comment:
Not entirely clear what "collection" means here, or there could be a more accurate term?

Reply:
Collection is just a vague description. Added more descriptions after this sentence in "main" branch.